### ISSUE TITLE
[BUG] Windows can't back to ssh linux because its trying to chnage permissions

### ISSUE THREAD

**Issue by guysoft** [rdiff-backup/rdiff-backup#486]
## Bug summary

tl;dr
Sync with windows to to linux on 2.0.5  gives the follwing.
This is after adding ``--no-acls`` and/or  ``--no-eas``

```
Sending back exception [Errno 1] Operation not permitted: b'/tmp/a' of type <class 'PermissionError'>:
  File "/usr/lib/python3.8/site-packages/rdiff_backup/connection.py", line 400, in answer_request
    result = eval(request.function_string)(*argument_list)
  File "/usr/lib/python3.8/site-packages/rdiff_backup/backup.py", line 270, in patch
    ITR.Finish()
  File "/usr/lib/python3.8/site-packages/rdiff_backup/rorpiter.py", line 277, in Finish
    to_be_finished.end_process()
  File "/usr/lib/python3.8/site-packages/rdiff_backup/backup.py", line 752, in end_process
    rpath.copy_attribs(self.dir_update, self.base_rp)
  File "/usr/lib/python3.8/site-packages/rdiff_backup/rpath.py", line 246, in copy_attribs
    rpout.chmod(rpin.getperms())
  File "/usr/lib/python3.8/site-packages/rdiff_backup/rpath.py", line 1138, in chmod
    self.conn.os.chmod(self.path,

Fatal Error: Lost connection to the remote system
```

## Version, Python, Operating System

On server:

```yaml
bash-5.0# rdiff-backup -v9
2020-11-11 13:00:15.806117 +0000  <CLIENT-149>  Fatal Error: No arguments given
See the rdiff-backup manual page for more information.
2020-11-11 13:00:15.671630 +0000  <CLIENT-149>  Using rdiff-backup version 2.0.5
2020-11-11 13:00:15.695974 +0000  <CLIENT-149>          with cpython /usr/bin/python3 version 3.8.5
2020-11-11 13:00:15.802587 +0000  <CLIENT-149>          on Linux-5.4.45-v8+-x86_64-with, fs encoding utf-8
```

On windows, using the exe provied in the release:
```
rdiff-backup.exe -v9
2020-11-11 15:01:50.386868 +0200  <CLIENT-428>  Using rdiff-backup version 2.0.5
2020-11-11 15:01:50.386868 +0200  <CLIENT-428>          with cpython C:\Users\user\Desktop\backupfriend-client\src\rdiff-backup.exe version 3.7.7
2020-11-11 15:01:50.386868 +0200  <CLIENT-428>          on Windows-10-10.0.18362, fs encoding utf-8
2020-11-11 15:01:50.386868 +0200  <CLIENT-428>  Fatal Error: No arguments given
See the rdiff-backup manual page for more information.
```

## rdiff-backup call

How did you call rdiff-backup, with which parameters:

In order to pass the paths correctly, I am using python's suprocess module, othweise providing the ``--remote-schema`` becomes a crasy mess.
The command is below, I can also provide a python script to reproduce:
```
ssh_path = r"C:\Users\user\Desktop\backupfriend-client\ssh.exe"
rdiff_path = rdiff_path = r"C:\Users\user\Desktop\backupfriend-client\src\rdiff-backup.exe"
rsa_path = r"c:\b\id_rsa"
cmd = [rdiff_path, "-v6", "--remote-schema", ssh_path + " -p 7022 -i " + rsa_path + " %s rdiff-backup --server", "--no-acls", "--no-eas",  "--", "C:\\a", "user@example.com::/tmp/a"]
```

## What happened and what did you expect?

The folder should sync and exit successfully.

## What happened
The folder synced fine but when arriving to the final stage it starts trying chmod and I get the error:

```
Sending back exception [Errno 1] Operation not permitted: b'/tmp/a' of type <class 'PermissionError'>:
  File "/usr/lib/python3.8/site-packages/rdiff_backup/connection.py", line 400, in answer_request
    result = eval(request.function_string)(*argument_list)
  File "/usr/lib/python3.8/site-packages/rdiff_backup/backup.py", line 270, in patch
    ITR.Finish()
  File "/usr/lib/python3.8/site-packages/rdiff_backup/rorpiter.py", line 277, in Finish
    to_be_finished.end_process()
  File "/usr/lib/python3.8/site-packages/rdiff_backup/backup.py", line 752, in end_process
    rpath.copy_attribs(self.dir_update, self.base_rp)
  File "/usr/lib/python3.8/site-packages/rdiff_backup/rpath.py", line 246, in copy_attribs
    rpout.chmod(rpin.getperms())
  File "/usr/lib/python3.8/site-packages/rdiff_backup/rpath.py", line 1138, in chmod
    self.conn.os.chmod(self.path,

Fatal Error: Lost connection to the remote system
```

## More information
1. I have chmod the sync folder destination to 777. I am using ``ssh.exe`` provided  in [OpenSSH](http://www.mls-software.com/opensshd.html) as recommended in the readme.
2. I moved the ``id_rsa`` key to ``C:\b`` folder because I have no idea how to esscape a folder with spaces in the current execution method with the remote schema. Help with that would also be great.

I am building a new open source sync program, that sits on your desktop and then syncs to your friends house your saved files.
Its going to be released open source along with a Raspberry Pi distro, also open source.
Client is WORK IN PROGRESS
Here:
https://github.com/guysoft/backupfriend-client

Docker server WORK IN PROGRESS
Here:
https://github.com/guysoft/BackupFriend-docker

It should work for window and linux (mac if someone wants to help me out).
In general I hope that kind of tool would make this amazing library more popular.

**Comment by guysoft** [comment]
Started working. I think restarting the remote server container fixed something. But can't really say more beyond "looks like permissions".

**Comment by ericzolf** [comment]
To answer your second question, `rsa_path = r'"c:\path with blanks\id_rsa"'` (double quotes within single quotes), should work.

Regarding the actual problem, I also had some strange behaviours regarding file system in a container. Anyway, let us know if you need more help, else close the ticket.

**Comment by guysoft** [comment]
What I am struggling now is a fairly strange situation, seems like wx.Execute is not returning the full output of rdiff-backup.
I am unsure how to reproduce it in a way that would be easy for you to test and run. I could try and arrange something up. But also considering just moving to subprocess module Popen.
It works fine in linux but not on windows.

**Comment by ericzolf** [comment]
I have strong doubts that the fact that `wx.Execute` doesn't show the full output of rdiff-backup has anything to do with rdiff-backup. At the end, rdiff-backup only writes to stdout/stderr and doesn't really care how it's called. Perhaps does `wx.Execute` not capture stderr on Windows, or perhaps it has a buffering issue, or perhaps you need to use [wx.Process](https://wxpython.org/Phoenix/docs/html/wx.Process.html).

**Comment by guysoft** [comment]
I think it could be a buffer issue. Are there any buffer settings?

> When I increase ther verbosity I get more lines. Ill try and make a
reproduce package if that helps.

**Comment by ericzolf** [comment]
Can you tell which output is missing more precisely?

**Comment by guysoft** [comment]
OK seems to be working. For some reason wx.execute logs the output fine but does not print it to the screen well, regardless I am closing this issue. Will ping rdiff-backup community when the software is usable.

**Comment by ericzolf** [comment]
Yep, sounds interesting. I close the issue as you didn't do it.
